File: src/codility/java/timecomplexity/TapeEquilibrium.java
package codility.java.timecomplexity;

public class TapeEquilibrium {
    public int solution(int[] A) {
        // write your code in Java SE 8
        int left = A[0];

        int right = 0;
        for (int i = 1; i < A.length; i++) {
            right += A[i];
        }

        int min = Math.abs(left - right);

        for (int i = 1; i < A.length - 1; i++) {
            left += A[i];
            right -= A[i];
            min = Math.min(min, Math.abs(left - right));
        }
        return min;
    }
}

File: src/codility/kotlin/TapeEquilibrium.kt
package codility.kotlin

import kotlin.math.absoluteValue

fun main() {
    TapeEquilibrium().solution(intArrayOf(-1,1))
}

class TapeEquilibrium {

    fun solution(A: IntArray): Int {
        // write your code in Kotlin
        var diff = Long.MAX_VALUE

        var left : Long = 0L
        var right : Long = A.map { it.toLong() }.sum()

        for (i in 0.. A.size -2) {
            left += A[i]
            right -= A[i]
            diff = Math.min(diff, (left - right).absoluteValue)
        }

        return diff.toInt()
    }

}